src/api/spec/controllers/webui/main_controller_spec.rb

```diff
@@ -70,4 +70,57 @@
       end
     end
   end
+
+  describe "GET #sitemap" do
+    render_views
+
+    before do
+      get :sitemap
+      @paths = Hash.from_xml(response.body)['sitemapindex']['sitemap'].map do |url|
+          uri = URI.parse(url['loc'])
+          "#{uri.path}?#{uri.query}"
+      end
+    end
+
+    it { expect(@paths).to include('/main/sitemap_projects?') }
+    it "have all category's urls" do
+      ('a'..'z').to_a.concat(('A'..'Z').to_a).each do |letter|
+        expect(@paths).to include("/main/sitemap_packages/show?category=home%3A#{letter}")
+      end
+    end
+    it { expect(@paths).to include("/main/sitemap_packages/show?category=opensuse") }
+  end
+
+  describe "GET #sitemap_projects" do
+    render_views
+
+    before do
+      create(:confirmed_user)
+      5.times { create(:project) }
```

Comment:
Just stumbled over [this](http://cookieshq.co.uk/posts/useful-factory-girl-methods/#but-i-need-them-in-large-numbers). So you probably can replace this with ```projects = create_list(:project, 5) ``` And use this array in your test instead of Project.all